cmd/all-in-one/main.go

```diff
@@ -271,48 +273,24 @@ func startQuery(
 	metricsQueryService querysvc.MetricsQueryService,
 	baseFactory metrics.Factory,
 	tm *tenancy.Manager,
+	jt *jtracer.JTracer,
 ) *queryApp.Server {
 	spanReader = storageMetrics.NewReadMetricsDecorator(spanReader, baseFactory.Namespace(metrics.NSOptions{Name: "query"}))
 	qs := querysvc.NewQueryService(spanReader, depReader, *queryOpts)
-	jtracer := jtracer.OT(opentracing.GlobalTracer())
-	server, err := queryApp.NewServer(svc.Logger, qs, metricsQueryService, qOpts, tm, jtracer)
+	server, err := queryApp.NewServer(svc.Logger, qs, metricsQueryService, qOpts, tm, jt)
 	if err != nil {
-		svc.Logger.Fatal("Could not start jaeger-query service", zap.Error(err))
+		svc.Logger.Fatal("Could not create jaeger-query", zap.Error(err))
 	}
 	go func() {
 		for s := range server.HealthCheckStatus() {
 			svc.SetHealthCheckStatus(s)
 		}
 	}()
 	if err := server.Start(); err != nil {
-		svc.Logger.Fatal("Could not start jaeger-query service", zap.Error(err))
+		svc.Logger.Fatal("Could not start jaeger-query", zap.Error(err))
 	}
-	return server
-}
 
-func initTracer(svc *flags.Service) io.Closer {
-	logger := svc.Logger
-	traceCfg := &jaegerClientConfig.Configuration{
-		ServiceName: "jaeger-query",
-		Sampler: &jaegerClientConfig.SamplerConfig{
-			Type:  "const",
-			Param: 1.0,
-		},
-		RPCMetrics: true,
-	}
-	traceCfg, err := traceCfg.FromEnv()
-	if err != nil {
-		logger.Fatal("Failed to read tracer configuration", zap.Error(err))
-	}
-	tracer, closer, err := traceCfg.NewTracer(
-		jaegerClientConfig.Metrics(jlibadapter.NewAdapter(svc.MetricsFactory)),
-		jaegerClientConfig.Logger(jaegerClientZapLog.NewLogger(logger)),
-	)
-	if err != nil {
-		logger.Fatal("Failed to initialize tracer", zap.Error(err))
```

Comment:
I think this is what it is. @albertteoh  we won't return here as well.

Reply:
Yeah, just questioning why; I just can't see a reason why the error shouldn't returned, but maybe there's a good reason.